### ISSUE TITLE
[Improvement]: AK/SK Leakage Security

### ISSUE THREAD

**Issue by shouwangyw** [apache/amoro#3310]
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

<img width="712" alt="image" src="https://github.com/user-attachments/assets/d09c3232-0139-4a61-8441-d52d66008aa2">
The rest api should probably not be whitelist, as this may expose the user's AK/SK, we need to think about a better way to provide OpenAPI.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by XBaith** [comment]
Thanks for reporting this issue. I reproduce this issue in my dev enviorment.
![image](https://github.com/user-attachments/assets/74966f1e-6af0-40bf-a372-964774844c29)

**Comment by XBaith** [comment]
Currently we can remove AK/SK in our configs and inject tempory iam credential via `vended-credentials`

**Comment by zhoujinsong** [comment]
The issue arises because the Iceberg Rest Catalog interface currently lacks authentication capabilities. However, if this interface has the potential to expose information from other catalogs at the moment, we should consider adding configuration to enable the Iceberg Rest Catalog, with the default setting being to keep it disabled until we implement an authentication mechanism for the Iceberg Rest Catalog interface.

HDYT? @shouwangyw @XBaith

**Comment by shouwangyw** [comment]
> The issue arises because the Iceberg Rest Catalog interface currently lacks authentication capabilities. However, if this interface has the potential to expose information from other catalogs at the moment, we should consider adding configuration to enable the Iceberg Rest Catalog, with the default setting being to keep it disabled until we implement an authentication mechanism for the Iceberg Rest Catalog interface.

Indeed, that's correct. In our usage, we remove these REST APIs from the whitelist and enforce Basic Auth and token validation for security.
